### PR TITLE
AB#90777 Add workflow update and tests for Environmental Impact

### PR DESCRIPTION
**Pull request by dawid-pawulski**

### CODE REVIEW

**Review by Carq — APPROVED** [review]

File: src/HE.Investment.AHP.WWW.Tests/Workflows/SiteWorkflowTests/CurrentStateTests.cs [hunk 1]
```diff
@@ -23,6 +23,8 @@ public class CurrentStateTests
 
     private readonly Section106Dto _section106 = new("3", "TestSite", false);
 
+    private EnvironmentalImpact? _environmentalImpact = new("reducing environmental impact");
```
Comment (Antek012) [inline src/HE.Investment.AHP.WWW.Tests/Workflows/SiteWorkflowTests/CurrentStateTests.cs:26]: Can we remove this field?
Idea of private Test method is to create complete, valid Site and use params for Test method to override some values to create specific test case.

File: src/HE.Investment.AHP.WWW.Tests/Workflows/SiteWorkflowTests/CurrentStateTests.cs [hunk 2]
```diff
@@ -269,6 +271,13 @@ public void ShouldReturnRuralClassification_WhenRuralClassificationNotProvided(b
         Test(SiteWorkflowState.RuralClassification, ruralClassification: new SiteRuralClassification(isWithinRuralSettlement, isRuralExceptionSite));
     }
 
+    [Fact]
+    public void ShouldReturnEnvironmentalImpact_WhenEnvironmentalImpactIsNotProvided()
+    {
+        _environmentalImpact = null;
+        Test(SiteWorkflowState.EnvironmentalImpact);
```
Comment (Antek012) [inline src/HE.Investment.AHP.WWW.Tests/Workflows/SiteWorkflowTests/CurrentStateTests.cs:278]: use Test method param.

**Review by Antek012 — APPROVED** [review]

File: src/HE.Investment.AHP.WWW.Tests/Workflows/SiteWorkflowTests/CurrentStateTests.cs (hunk 1 above)
Comment (dawid-pawulski) [reply src/HE.Investment.AHP.WWW.Tests/Workflows/SiteWorkflowTests/CurrentStateTests.cs:26]: After the discussion, we leave it as it is for now.

File: src/HE.Investment.AHP.WWW.Tests/Workflows/SiteWorkflowTests/CurrentStateTests.cs (hunk 2 above)
Comment (dawid-pawulski) [reply src/HE.Investment.AHP.WWW.Tests/Workflows/SiteWorkflowTests/CurrentStateTests.cs:278]: After the discussion, we leave it as it is for now.

**Review by jjankowski87 — APPROVED** [review]

**Pull request merged**